### PR TITLE
[TC22-1274] Fix employment fields confusion

### PR DESCRIPTION
**Pull request by kalmerkaurson**

### CODE REVIEW

File: affiliates/mexico.swagger.json [hunk 1]
```diff
@@ -420,11 +420,13 @@
           },
           "employed_since" : {
             "type" : "string",
-            "example" : "yyyy-mm-dd"
+            "example" : "yyyy-mm-dd",
```
Comment (Nuffic) [inline affiliates/mexico.swagger.json]: I doubt this example will pass as a value

Comment (Nuffic) [reply affiliates/mexico.swagger.json]: This could be added as description

Comment (kalmerkaurson) [reply affiliates/mexico.swagger.json]: fixed date examples

File: affiliates/mexico.swagger.json [hunk 2]
```diff
@@ -416,15 +416,17 @@
           },
           "remuneration_deadline" : {
             "type" : "string",
-            "example" : "yyyy-mm-dd"
+            "example" : "2023-04-18"
```
Comment (mikk150) [inline affiliates/mexico.swagger.json]: but what's the format?

**Review by Nuffic — APPROVED** [review]

File: affiliates/mexico.swagger.json [hunk 3]
```diff
@@ -416,15 +416,17 @@
...
           },
           "employed_since" : {
             "type" : "string",
-            "example" : "yyyy-mm-dd"
+            "example" : "2021-02-18",
```
Comment (mikk150) [inline affiliates/mexico.swagger.json:427]: but what's the format?

File: affiliates/mexico.swagger.json [hunk 4]
```diff
@@ -342,7 +342,7 @@
           },
           "birth_date" : {
             "type" : "string",
-            "description" : "yyyy-mm-dd"
+            "description" : "1989-05-23"
```
Comment (mikk150) [inline affiliates/mexico.swagger.json]: but what's the format?

**Review by mikk150 — CHANGES_REQUESTED** [review]

File: affiliates/mexico.swagger.json (hunk 2 above)
Comment (kalmerkaurson) [reply affiliates/mexico.swagger.json]: added format description

File: affiliates/mexico.swagger.json (hunk 3 above)
Comment (kalmerkaurson) [reply affiliates/mexico.swagger.json:427]: added format description

File: affiliates/mexico.swagger.json (hunk 4 above)
Comment (kalmerkaurson) [reply affiliates/mexico.swagger.json]: added format description

**Review by Nuffic — APPROVED** [review]

File: affiliates/mexico.swagger.json (hunk 2 above)
Comment (mikk150) [reply affiliates/mexico.swagger.json]: according to json format validator, there is `format` field, why don't we use that?

Comment (kalmerkaurson) [reply affiliates/mexico.swagger.json]: now using "date" format and regex pattern only allows 1000-2999 for year

**Review by mikk150 — APPROVED** [review]

**Pull request merged**